docs/orchestration/execution/storage_options.md

```diff
@@ -140,6 +140,59 @@ If you do not specify a `registry_url` for your Docker Storage then the image wi
 Docker Storage uses the [Docker SDK for Python](https://docker-py.readthedocs.io/en/stable/index.html) to build the image and push to a registry. Make sure you have the Docker daemon running locally and you are configured to push to your desired container registry. Additionally make sure whichever platform Agent deploys the container also has permissions to pull from that same registry.
 :::
 
+## Webhook
+
+[Webhook Storage](/api/latest/environments/storage.html#webhook) is a storage option that stores and retrieves flows with HTTP requests. This type of storage can be used with any type of agent, and is intended to be a flexible way to integrate Prefect with your existing ecosystem, including your own file storage services.
+
+For example, the following code could be used to store flows in DropBox.
+
+```python
+from prefect import Flow
+from prefect.environments.storage import Webhook
+
+flow = Flow(
+    "dropbox-flow",
+    storage=Webhook(
+        build_kwargs={
+            "url": "https://content.dropboxapi.com/2/files/upload",
+            "headers": {
+                "Content-Type": "application/octet-stream",
+                "Dropbox-API-Arg": json.dumps(
+                    {
+                        "path": "/Apps/prefect-test-app/dropbox-flow.flow",
+                        "mode": "overwrite",
+                        "autorename": False,
+                        "strict_conflict": True,
+                    }
+                ),
+            },
+        },
+        build_http_method="POST",
+        get_flow_kwargs={
+            "url": "https://content.dropboxapi.com/2/files/download",
+            "headers": {
+                "Accept": "application/octet-stream",
+                "Dropbox-API-Arg": json.dumps(
+                    {"path": "/Apps/prefect-test-app/dropbox-flow.flow"}
+                ),
+            },
+        },
+        get_flow_http_method="POST",
+        build_secret_config={
```

Comment:
I also find the `get_flow` and `build` prefixes on these kwargs a bit off. I know they correspond to the requests for the `build` and `get_flow` methods, but without the word `request` in there `build_kwargs` looks like kwargs to pass to `build` to me. Feels too tied to the interface method names and not tied to what the requests are actually doing (storing and loading bytes). Perhaps?

```
store_request_kwargs=...,
store_request_method=...,
load_request_kwargs=...,
load_request_method=...,
```

I'd use `put` and `get`, except those conflict with the http methods. Not a strong opinion, just a thought.

Reply:
I thought this was sufficiently complex that I shouldn't make it part of the first pass, but if you think it's necessary I'm happy to add it!

I'm a little worried about free-form templating everything though...that's going to be a problem if you have JSON jammed in a string, like the DropBox API requires (https://github.com/jameslamb/webhook-storage-integration-tests/blob/3bc93bf2ce4b9a0539306045f2f6a82bc3325c53/test-dropbox.py#L47). That opens you up to needing to know how to escape the right `}`, which doesn't sound fun.

maybe it would be simpler to, instead of templating individual string fields, just allow people to replace the *entire value* of any build_kwarg or get_flow_kwarg with the value of an environment variable / secret?

>  what if you need to modify the secret before transmitting (e.g. adding a Bearer prefix))

I did think about this specific case. If your API token's literal value is `abc`, there's no reason you couldn't put `Bearer abc` into an environment variable / secret, right? Without needing to have any code run to add `Bearer` to the front.

Reply:
It feels non-intuitive to me to require storing a full authorization header/url/etc... in a secret to make proper use of it. If we keep the `$` prefix requirement that `string.Template` uses, that (I believe) avoids the issue of accidentally templating things that just happen to contain `{}` characters, since they won't match the regex.

Reply:
ooooo ook! I like that a lot. I'll add that to this PR.

As for the names, I feel that there's value in coupling to the method names actually. `build()` and `get_flow()` are important to understand when using a Storage object, I think, and I'd rather couple to those than invent another thing people have to reason about. But I do like adding `_request` to make that clearer. How do you feel about `build_request_*`and `get_flow_request_*`?

Reply:
>  But I do like adding `_request` to make that clearer.

Makes sense to me!

Reply:
Ok I just attempted the templating thing. Awesome suggestion, I like this a lot better than the `secret_config` approach.

Commit: https://github.com/PrefectHQ/prefect/pull/3000/commits/937dd84700ba5799319da2637a593d4a61cbbf64

I also updated my integration tests and ran them to be sure it's working as expected: https://github.com/jameslamb/webhook-storage-integration-tests.

## Note for reviewers

I think it could be valuable to offer more explicit control over whether environment variables or Prefect secrets are used, to avoid issues caused by conflicting names.

I think that could be done in a backwards-compatible way in the future, by adding a `render_preferences` argument that is like `{"SOME_VARIABLE": "environment"}`,  which changes the behavior for rendering `${SOME_VARIABLE}` from "env --> secret --> error-if-absent" to "env --> error-if-absent".

I thought that complexity wasn't worth it for a first pass, but I'd be happy to add something like it if reviewers think it's a good idea.